util/array.h

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provide fixed-width array implementation for usage in kernel functions
+//  because std::array methods are not required to be noexcept by C++ spec,
+//  while exceptions are not allowed in a device function
+//
+*******************************************************************************/
+
+#ifndef __SYCLCTS_UTIL_ARRAY_H
+#define __SYCLCTS_UTIL_ARRAY_H
+
+namespace sycl_cts {
+namespace util {
+
+/** @brief Fixed width array to use in kernel functions
+ */
+template <class T, std::size_t N>
+struct array {
+  // Should be aggregate type, so no explicit constructors
+  // and private attributes
+
+  using value_type = T;
+  using reference = value_type&;
+  using const_reference = const value_type&;
+  using size_type = std::size_t;
+
+  /** @brief Iterator type; is safe to use on host or device side only
+   */
+  using iterator = value_type*;
+  using const_iterator = const iterator;
+
+  /** @brief Internal data storage;
+   *         public for aggregate initialization availability
+   */
+  value_type values[N];
+
+  reference operator[](size_type pos) {
+    return values[pos];
+  }
+  const_reference operator[](size_type pos) const {
+    return values[pos];
+  }
+
+  inline constexpr size_type size() const noexcept {
```

Comment:
```suggestion
  constexpr size_type size() const noexcept {
```

`constexpr` implies `inline` when used on functions.